sale_order_wh_from_delivery_carrier/models/sale_order.py

```diff
@@ -0,0 +1,14 @@
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends("user_id", "company_id", "carrier_id")
+    def _compute_warehouse_id(self):
+        # Set warehouse by shipping method
+        so_with_shipping = self.filtered("carrier_id")
+        for order in so_with_shipping:
+            if order.state == "draft" and order.carrier_id.so_warehouse_id:
```

Comment:
computer method uses in its calculations also the `state` field of the order, please add it to api.depends

Reply:
Resolved